### PR TITLE
feat(aws-lambda): integration edge cases

### PR DESCRIPTION
**Pull request by scefali**
This PR adds handling for the following edge cases for the AWS Lambda integration
1. User has one project: automatically use that project and skip the project selection step
2. Validate ARN
3. Moves externalID generation to the cloudformation page in case we skip the first step when the user has exactly one project.

Note this PR exposes limitations in the current React pipeline approach:
1. In order to use form submits, we have to first redirect the popup from `/${orgSlug}/integrations/aws_lambda/setup/` to `/extensions/aws_lambda/setup/`. This is because a form submit will reload the current URL with a `GET` request after the submit if we provide the `skipPreventDefault` prop which is needed to load the next view after a submission. The problem is that doing a `GET` request to `/${orgSlug}/integrations/aws_lambda/setup/` will regenerate the pipeline while a GET request to`/extensions/aws_lambda/setup/` will maintain the same pipeline state. The workaround is to make the first step in the pipeline (which could be the project select or CloudFormation step depending on if we skip the first step if the user has exactly one project) redirect to `/extensions/aws_lambda/setup/` and add the form fields to the query parameters. In those views, we simulate a form `POST` by generating a toast and calling `model.setFormSaving()`. It's kind of hacky but it works.
2. Handling errors on the backend is a pain. Because the page will reload after a redirect, we will lose the current form field values without injecting them back in which is a hack.

Recommendation: if we need new installation flows that are somewhat complicated (multiple steps that require user input), we should look into a pipeline with a JSON interface. In that flow, we would return JSON instead of an HTML view with the API and let the browser advance to the next view when the API returns successfully.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaCloudformation.tsx [hunk 1]
```diff
@@ -1,30 +1,79 @@
...
+  error?: string;
 };
 
 export default class AwsLambdaCloudformation extends React.Component<Props> {
-  model = new FormModel({apiOptions: {baseUrl: window.location.origin}});
+  componentDidMount() {
+    // show the error if we have it
+    const {error} = this.props;
+    if (error) {
+      addErrorMessage(error, {duration: 10000});
```
Comment (NisanthanNanthakumar) [inline src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaCloudformation.tsx:39]: @scefali I think we should add the translation `tct` for the error message

File: src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaCloudformation.tsx [hunk 2]
```diff
@@ -1,30 +1,79 @@
...
+    addLoadingMessage(t('Submitting\u2026'));
+    this.model.setFormSaving();
+    const {
+      location: {origin},
+    } = window;
+    // redirect to the extensions endpoint with the form fields as query params
+    // this is needed so we don't restart the pipeline loading from the original
+    // OrganizationIntegrationSetupView route
+    const newUrl = `${origin}/extensions/aws_lambda/setup/?${qs.stringify(data)}`;
+    window.location.replace(newUrl);
```
Comment (NisanthanNanthakumar) [inline src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaCloudformation.tsx:73]: @scefali this view is in the popup, right? Doing `window.location.replace()` will prevent the user from using the back button. Maybe we can use `window.location.assign()`? Or also can't we use React Router?

File: src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaProjectSelect.tsx [hunk 3]
```diff
@@ -1,51 +1,52 @@
...
+      addLoadingMessage(t('Submitting\u2026'));
+      this.model.setFormSaving();
+      const {
+        location: {origin},
+      } = window;
+      // redirect to the extensions endpoint with the projectId as a query param
+      // this is needed so we don't restart the pipeline loading from the original
+      // OrganizationIntegrationSetupView route
+      const newUrl = `${origin}/extensions/aws_lambda/setup/?projectId=${projectId}`;
+      window.location.replace(newUrl);
```
Comment (NisanthanNanthakumar) [inline src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaProjectSelect.tsx:41]: @scefali same question here

File: src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaCloudformation.tsx (hunk 1 above)
Comment (scefali) [reply src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaCloudformation.tsx:39]: @NisanthanNanthakumar You can't actually do translations with dynamic text like that but I can put the translations on the backend for the error message string.

File: src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaCloudformation.tsx (hunk 2 above)
Comment (scefali) [reply src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaCloudformation.tsx:73]: @NisanthanNanthakumar `window.location.assign()` works, I'll switch to that

**Review by NisanthanNanthakumar — APPROVED** [review]

**Pull request merged**